loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+import { Nullable } from "babylonjs/types";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Mesh } from 'babylonjs/Meshes/mesh';
+import { INode, IAccessor } from "../glTFLoaderInterfaces";
+import { IGLTFLoaderExtension } from "../glTFLoaderExtension";
+import { GLTFLoader, ArrayItem } from "../glTFLoader";
+import { Vector3, Quaternion } from 'babylonjs/Maths/math.vector';
+
+const NAME = "KHR_mesh_instancing";
+
+interface IKHRMeshInstancing {
+    mesh?: number;
+    attributes: { [name: string]: number };
+}
+
+/**
+ * [Proposed Specification](https://github.com/KhronosGroup/glTF/pull/1691)
+ * [Playground Sample](//TODO)
+ * !!! Experimental Extension Subject to Changes !!!
+ */
+export class KHR_mesh_instancing implements IGLTFLoaderExtension {
+    /**
+     * The name of this extension.
+     */
+    public readonly name = NAME;
+
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /** @hidden */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public loadNodeAsync(context: string, node: INode, assign: (babylonTransformNode: TransformNode) => void): Nullable<Promise<TransformNode>> {
+        return GLTFLoader.LoadExtensionAsync<IKHRMeshInstancing, TransformNode>(context, node, this.name, (extensionContext, extension) => {
+            const nodeMesh = node.mesh;
+            const promise = this._loader.loadNodeAsync(`#/nodes/${node.index}`, node, (babylonTransformNode) => {
+                babylonTransformNode.parent = this._loader.rootBabylonMesh;
+                const attributeBufferPromises = new Array<Promise<any>>();
+
+                let attributes = extension.attributes;
+                let attributeKeys = Object.keys(attributes);
+                const attributeBuffers : { [name: string]: { buffer : Float32Array, accessor : IAccessor } } = {};
+                for (let i = 0; i < attributeKeys.length; ++i) {
+                    let attributeKey = attributeKeys[i];
+                    let accessor = ArrayItem.Get(`${extensionContext}/attributes/${attributeKey}`, this._loader.gltf.accessors, attributes[attributeKey]);
+                    attributeBufferPromises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attributeKey] = { buffer : data, accessor : accessor };
+                    }));
+                }
```

Comment:
I would recommend not looping through all the attributes since we will only be able to understand specific ones and thus loading them will be pointless. I would explicitly load `TRANSLATION`, `ROTATION`, `SCALE`.

Reply:
Ok, I created a helper function to handle the asynchronous attribute retrieval, and we only explicitly retrieve `TRANSLATION`, `ROTATION` and `SCALE`